### PR TITLE
Sfl charity preference

### PR DESCRIPTION
**Pull request by aniket-Webonise**

### CODE REVIEW

File: app/src/main/java/com/sjana/fitness/model/CharityPreferenceRequest.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+data class charityPreference(
+    @SerializedName("charity1")
+    val charity1: Boolean?,
+    @SerializedName("charity2")
+    val charity2: Boolean? ,
+    @SerializedName("charity3")
+    val charity3: Boolean?
+
+)
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/model/CharityPreferenceRequest.kt:18]: EOL

File: app/src/main/java/com/sjana/fitness/model/CharityResponseModel.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+data class latestCharityPreference (
+    @SerializedName("charity1")
+    val charity1: Boolean?,
+    @SerializedName("charity2")
+    val charity2: Boolean? ,
+    @SerializedName("charity3")
+    val charity3: Boolean?
+)
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/model/CharityResponseModel.kt:17]: EOL

File: app/src/main/java/com/sjana/fitness/model/UserAttributes.kt [hunk 3]
```diff
@@ -28,8 +28,23 @@ data class UserAttributes(
...
+    @SerializedName("latest_charity_preference")
+    var latestChariTyPreference : LatestChariTyPreference=LatestChariTyPreference()
+
     //add-ons
     var journeyAttributes: JourneyAttributes = JourneyAttributes()
     var userPrefAttributes: UserPrefAttributes = UserPrefAttributes()
     var valueCards: List<ValueCardsUserData> = listOf()
 }
+
+data class LatestChariTyPreference(
```
Comment (tejasAlsi-Webonise) [inline app/src/main/java/com/sjana/fitness/model/UserAttributes.kt:41]: charity preference is not a user attribute, keep this object light. fetch the charity preference from API

File: app/src/main/res/layout/charity_preference_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            android:layout_height="wrap_content"
+            android:drawableLeft="@drawable/radio_drawable"
+            android:drawablePadding="8dp"
+            android:button="@null"
+            android:checked="true"
+            android:layout_marginTop="10dp"
+            android:fontFamily="@font/heebo_bold"
+            android:text="@string/world_vision_society"
+            android:textColor="@color/darkTxtColor"
+            android:textSize="14sp" />
```
Comment (tejasAlsi-Webonise) [inline app/src/main/res/layout/charity_preference_dialog.xml:80]: this is wrong, use radio group button.

**Review by tejasAlsi-Webonise — CHANGES_REQUESTED** [review]
check comments

File: app/src/main/res/layout/charity_preference_dialog.xml (hunk 4 above)
Comment (aniket-Webonise) [reply app/src/main/res/layout/charity_preference_dialog.xml:80]: requirement is different its not like radio so checkbox is used

File: app/src/main/java/com/sjana/fitness/model/UserAttributes.kt (hunk 3 above)
Comment (aniket-Webonise) [reply app/src/main/java/com/sjana/fitness/model/UserAttributes.kt:41]: done

**Review by tejasAlsi-Webonise — APPROVED** [review]

**Pull request merged**